### PR TITLE
Added showVelocity

### PR DESCRIPTION
**Pull request by NekoRem**
Could have been done better but this works fine...
This renders your velocity onscreen which i guess can be useful for movement players ¯\_(ツ)_/¯

### CODE REVIEW

**Comment by Cyk-Fad** [comment]
Maybe......

**Comment by koahook** [comment]
Thank you so much i was trying to use your code last night and just couldnt make it work

**Comment by aljawary** [comment]
Ss?

**Comment by NekoRem** [comment]
> Ss?

image

**Comment by NekoRem** [comment]
i might make it bigger or match viewmodel/crosshair colour

**Comment by danielkrupinski** [comment]
Why not **cl_showpos 1**?

**Comment by koahook** [comment]
> Why not **cl_showpos 1**?

people like it at the bottom for some reason

**Comment by NekoRem** [comment]
> Why not **cl_showpos 1**?

Personally i don't like having the decimal places. I also like to have it in a place where i can see it easily with peripheral vision

File: Osiris/Hacks/Visuals.cpp [hunk 1]
```diff
@@ -393,3 +399,25 @@ void Visuals::skybox() noexcept
...
+{
+    return strconverter.from_bytes(str);
+}
+
+void Visuals::showVelocity() noexcept
+{
+    if (!config->visuals.showvelocity || !localPlayer)
+        return;
+
+    double velocity = localPlayer->velocity().length2D();
```
Comment (notgoodusename) [inline Osiris/Hacks/Visuals.cpp:413]: length2D returns a float, not a double

File: Osiris/Hacks/Visuals.cpp [hunk 2]
```diff
@@ -393,3 +399,25 @@ void Visuals::skybox() noexcept
...
+    std::stringstream velocitystream;
+    velocitystream << std::fixed << std::setprecision(0) << velocity;
+    std::wstring velocitywstr = towstring(velocitystream.str());
```
Comment (notgoodusename) [inline Osiris/Hacks/Visuals.cpp:416]: can be done better
`std::wstring velocitywstr{ L"Velocity: " + std::to_wstring(static_cast<float>(velocity)) };`
can change from float to int, if you want to remove decimals

File: Osiris/Hacks/Visuals.cpp [hunk 3]
```diff
@@ -393,3 +399,25 @@ void Visuals::skybox() noexcept
...
+
+    interfaces->surface->setTextFont(Surface::font);
+    interfaces->surface->setTextColor(255, 255, 255);
+    const auto [width, height] = interfaces->surface->getScreenSize();
+    interfaces->surface->setTextPosition(width / 2 - 6, height - 200);
```
Comment (notgoodusename) [inline Osiris/Hacks/Visuals.cpp:421]:
```
const auto velocityWidth = interfaces->surface->getTextSize(Surface::font, velocitywstr.c_str()).first;
interfaces->surface->setTextPosition(width / 2 - velocityWidth, height - 200);
```

File: Osiris/Hacks/Visuals.cpp [hunk 4]
```diff
@@ -393,3 +399,25 @@ void Visuals::skybox() noexcept
...
+    if (!config->visuals.showvelocity || !localPlayer)
+        return;
+
+    double velocity = localPlayer->velocity().length2D();
+    std::stringstream velocitystream;
+    velocitystream << std::fixed << std::setprecision(0) << velocity;
+    std::wstring velocitywstr = towstring(velocitystream.str());
+
+    interfaces->surface->setTextFont(Surface::font);
+    interfaces->surface->setTextColor(255, 255, 255);
```
Comment (notgoodusename) [inline Osiris/Hacks/Visuals.cpp:419]: Could use colorpicker for this

File: Osiris/Hacks/Visuals.cpp [hunk 5]
```diff
@@ -393,3 +399,25 @@ void Visuals::skybox() noexcept
         memory->loadSky(sv_skyname->string);
     }
 }
+
+std::wstring towstring(std::string str)
+{
+    return strconverter.from_bytes(str);
+}
+
+void Visuals::showVelocity() noexcept
```
Comment (notgoodusename) [inline Osiris/Hacks/Visuals.cpp:408]: Why do it in visuals? Misc would be the place too put this

**Comment by koahook** [comment]
crashes when enabled \o_o/

**Comment by NekoRem** [comment]
> crashes when enabled \o_o/

Probably ive not tried it

**Comment by koahook** [comment]
> > crashes when enabled \o_o/
> 
> Probably ive not tried it

how'd you get the screenshot then?

**Comment by newcommerdontblame** [comment]
what a useless pr

**Comment by NekoRem** [comment]
> what a useless pr

Thank you

**Comment by NekoRem** [comment]
> > > crashes when enabled \o_o/
> > 
> > 
> > Probably ive not tried it
> 
> how'd you get the screenshot then?

it works perfectly on my personal source of osiris

File: Osiris/Hacks/Visuals.cpp (hunk 5 above)
Comment (NekoRem) [reply Osiris/Hacks/Visuals.cpp:397]: It renders text on the screen so i believe it belongs in visuals

File: Osiris/Hacks/Visuals.cpp (hunk 3 above)
Comment (NekoRem) [reply Osiris/Hacks/Visuals.cpp:413]: done

File: Osiris/Hacks/Visuals.cpp (hunk 2 above)
Comment (NekoRem) [reply Osiris/Hacks/Visuals.cpp]: done thanks

File: Osiris/Hacks/Visuals.cpp (hunk 1 above)
Comment (NekoRem) [reply Osiris/Hacks/Visuals.cpp]: indeed it does sorry

**Comment by koahook** [comment]
> > > > crashes when enabled \o_o/
> > > 
> > > 
> > > Probably ive not tried it
> > 
> > 
> > how'd you get the screenshot then?
> 
> it works perfectly on my personal source of osiris

just built your source and it still crashes

**Comment by NekoRem** [comment]
Does it crash fir anybody else?

File: Osiris/Hacks/Visuals.cpp (hunk 3 above)
Comment (notgoodusename) [reply Osiris/Hacks/Visuals.cpp:412]: you didnt do it

File: Osiris/Hacks/Visuals.cpp [hunk 6]
```diff
@@ -393,3 +393,22 @@ void Visuals::skybox() noexcept
...
+{
+    if (!config->visuals.showvelocity.enabled || !localPlayer->isAlive())
+        return;
+
+    float velocity = localPlayer->velocity().length2D();
+    std::wstring velocitywstr{ L"Velocity: " + std::to_wstring(static_cast<int>(velocity)) };
+
+    interfaces->surface->setTextFont(Surface::font);
+    if (config->visuals.showvelocity.rainbow)
+        interfaces->surface->setTextColor(rainbowColor(memory->globalVars->realtime, config->visuals.showvelocity.rainbowSpeed));
```
Comment (notgoodusename) [inline Osiris/Hacks/Visuals.cpp:407]: ` interfaces->surface->setTextColor(rainbowColor(config->visuals.showvelocity.rainbowSpeed));`
Should fix the compilation error

File: Osiris/Hacks/Visuals.cpp [hunk 7]
```diff
@@ -400,23 +394,20 @@ void Visuals::skybox() noexcept
...
 
-std::wstring towstring(std::string str)
-{
-    return strconverter.from_bytes(str);
-}
-
 void Visuals::showVelocity() noexcept
 {
-    if (!config->visuals.showvelocity || !localPlayer)
+    if (!config->visuals.showvelocity.enabled || !localPlayer->isAlive())
```
Comment (notgoodusename) [inline Osiris/Hacks/Visuals.cpp:399]: you forgot to check if !localPlayer

File: Osiris/Hacks/Visuals.cpp (hunk 6 above)
Comment (NekoRem) [reply Osiris/Hacks/Visuals.cpp:407]: Ok i will fix it but i swear it compiled fine for me

File: Osiris/Hacks/Visuals.cpp (hunk 7 above)
Comment (NekoRem) [reply Osiris/Hacks/Visuals.cpp:399]: Ah yes that would be why

Comment (koahook) [reply Osiris/Hacks/Visuals.cpp:399]: > Ah yes that would be why

still is crashing after enabling feature thought

**Comment by NekoRem** [comment]
Sorry i don't do pull requests often

**Comment by senisditron** [comment]
how to center it?

Bez tytułu

**Comment by koahook** [comment]
its probably centered but csgo’s player count/indicator isnt check it up clock next to the crosshair

**Comment by danielkrupinski** [comment]
@senisditron You can subtract half of text width from x position.

**Comment by NekoRem** [comment]
What daniel said

**Comment by koon666** [comment]
does anyone know the code that takes the last unit the player in jump?
I'm looking for this code for a long time, but I never think ;(

Example: https://prnt.sc/tgir6s

**Comment by NekoRem** [comment]
not bothered to keep fixing this just paste it in yourself

**Comment by braken1** [comment]
hey, thanks it works for me. Do you know how to make outline on text?

**Comment by NekoRem** [comment]
> hey, thanks it works for me. Do you know how to make outline on text?

Think you can show me an example of what you mean?

**Comment by braken1** [comment]
> > hey, thanks it works for me. Do you know how to make outline on text?
> 
> Think you can show me an example of what you mean?

image

this black shadow behind the velocity like so it would be more visible

**Comment by braken1** [comment]
> > > hey, thanks it works for me. Do you know how to make outline on text?
> > 
> > 
> > Think you can show me an example of what you mean?
> 
> image
> 
> this black shadow behind the velocity like so it would be more visible

image

**Comment by shakeit1337** [comment]
how to add this line by velocity have one the code for this ?

> > > > hey, thanks it works for me. Do you know how to make outline on text?
> > > 
> > > 
> > > Think you can show me an example of what you mean?
> > 
> > 
> > image
> > this black shadow behind the velocity like so it would be more visible
> 
> image

**Comment by shakeit1337** [comment]
> > > hey, thanks it works for me. Do you know how to make outline on text?
> > 
> > 
> > Think you can show me an example of what you mean?
> 
> image
> 
> this black shadow behind the velocity like so it would be more visible

how to add this line by velocity ? have one the code for this ?

**Comment by NekoRem** [comment]
> > > > hey, thanks it works for me. Do you know how to make outline on text?
> 
> > > 
> 
> > > 
> 
> > > Think you can show me an example of what you mean?
> 
> > 
> 
> > image
> 
> > 
> 
> > this black shadow behind the velocity like so it would be more visible
> 
> 
> 
> how to add this line by velocity ? have one the code for this ?
> 
> 

This is closed. Lets kindly leave it that way

**Pull request closed without merging**